app/models/threads.js

```diff
@@ -308,12 +308,45 @@
             if (updates.distribution && 'expression' in updates.distribution &&
                 updates.distribution.expression != this.get('distribution')) {
                 const dist = updates.distribution.expression;
+                const oldDist = this.get('distribution');
                 const normalized = await F.atlas.resolveTagsFromCache(dist);
+                const oldNormalized = await F.atlas.resolveTagsFromCache(oldDist); 
                 if (normalized.universal !== updates.distribution.expression) {
                     F.util.reportError('Non-universal expression sent by peer',
                                        {distribution: dist});
                 }
-                this.addNotice("Distribution Changed", normalized.pretty);
+                let diff;
+                let added;
+                var newIncludedTags = new F.util.ESet(normalized.includedTagids);                
+                var oldIncludedTags = new F.util.ESet(oldNormalized.includedTagids);
+                var newExcludedTags = new F.util.ESet(normalized.excludedTagids);                
+                var oldExcludedTags = new F.util.ESet(oldNormalized.excludedTagids);
+                if (newIncludedTags.size > oldIncludedTags.size) {
```

Comment:
Since there could be any mixture of added removals, removed removals, added additions, and removed additions, your pattern here of assuming only one of those could happen will miss a lot of cases. Maybe consider accumulating any additions (removed removals, added additions) and any removals (removed additions, added removals) so you can show all of what might have happened...

Reply:
You are still going to miss important cases because a set may not change size while still gaining and losing members.

Here's the set math to accumulate your overall "added" and "removed" tag sets:
* added = (oldExcluded - newExcluded) + (newIncluded - oldIncluded)
* removed = (newExcluded - oldExcluded) + (oldIncluded - newIncluded)